src/main/scala/io/scalac/tezos/translator/routes/dto/LibraryEntryRoutesDto.scala

```diff
@@ -11,24 +13,33 @@ case class LibraryEntryRoutesDto(
   micheline: String,
   michelson: String
 ) {
-  def toDomain: LibraryEntry = LibraryEntry(
-    uid = Uid(),
-    name = name,
-    author = author,
-    email = email,
-    description = description,
-    micheline = micheline,
-    michelson = michelson,
-    status = PendingApproval
-  )
+  def toDomain: Try[LibraryEntry] = {
+    val emailAddress = email match {
+      case Some(v) => EmailAddress.fromString(v).map(Some(_))
+      case None => Success(None)
+    }
+
+    emailAddress.map { ea =>
+      LibraryEntry(
+        uid = Uid(),
+        name = name,
+        author = author,
+        email = ea,
+        description = description,
+        micheline = micheline,
+        michelson = michelson,
+        status = PendingApproval
+      )
+    }
+  }
 }
 
 object LibraryEntryRoutesDto {
   def fromDomain(v: LibraryEntry): LibraryEntryRoutesDto =
     LibraryEntryRoutesDto(
       name = v.name,
       author = v.author,
-      email = v.email,
+      email = v.email.map(_.toString),
```

Comment:
without email as an option you could overwrite toString to and change this into v.email.toString